diff --git a/dist/turbo.es2017-umd.js b/dist/turbo.es2017-umd.js
index 4d6afbcbb3c42cba497b80848cb0ac4052ac629a..174edec13a492e0c56363f478878123dd965d2c3 100644
--- a/dist/turbo.es2017-umd.js
+++ b/dist/turbo.es2017-umd.js
@@ -2636,7 +2636,7 @@ Copyright © 2024 37signals LLC
           this.scrollToAnchor() || this.view.scrollToTop();
         }
         if (this.isSamePage) {
-          this.delegate.visitScrolledToSamePageLocation(this.view.lastRenderedLocation, this.location);
+          this.delegate.visitScrolledToSamePageLocation(window.location.href, this.location);
         }
 
         this.scrolled = true;
@@ -3382,13 +3382,12 @@ Copyright © 2024 37signals LLC
 
     locationWithActionIsSamePage(location, action) {
       const anchor = getAnchor(location);
-      const currentAnchor = getAnchor(this.view.lastRenderedLocation);
       const isRestorationToTop = action === "restore" && typeof anchor === "undefined";
 
       return (
         action !== "replace" &&
         getRequestURL(location) === getRequestURL(this.view.lastRenderedLocation) &&
-        (isRestorationToTop || (anchor != null && anchor !== currentAnchor))
+        (isRestorationToTop || anchor != null)
       )
     }
 
