diff --git a/core/misc/displace.js b/core/misc/displace.js
index 3032db7af7a..73084be85fc 100644
--- a/core/misc/displace.js
+++ b/core/misc/displace.js
@@ -159,8 +159,12 @@
     const n = displacingElements.length;
     for (let i = 0; i < n; i++) {
       const el = displacingElements[i];
-      // If the element is not visible, do consider its dimensions.
-      if (el.style.display === 'none') {
+      // If the element is set to display: none; do not consider its dimensions.
+      //
+      // @see https://api.jquery.com/hidden-selector/
+      //   For a more thorough implementation that considers other properties
+      //   and whether the element has layout at all.
+      if (getComputedStyle(el).display === 'none') {
         continue;
       }
       // If the offset data attribute contains a displacing value, use it.
