diff --git a/core/lib/Drupal/Core/Asset/JsCollectionGrouper.php b/core/lib/Drupal/Core/Asset/JsCollectionGrouper.php
index 17253626f2..6b76ad74c7 100644
--- a/core/lib/Drupal/Core/Asset/JsCollectionGrouper.php
+++ b/core/lib/Drupal/Core/Asset/JsCollectionGrouper.php
@@ -33,6 +33,13 @@ public function group(array $js_assets) {
           // Help ensure maximum reuse of aggregate files by only grouping
           // together items that share the same 'group' value.
           $group_keys = $item['preprocess'] ? [$item['type'], $item['group']] : FALSE;
+          if (!empty($item['attributes'])) {
+            $attributes = array_filter($item['attributes']);
+            if (!empty($attributes)) {
+              $attribute_groups = array_keys($attributes);
+              $group_keys = !empty($group_keys) ? array_merge($group_keys, $attribute_groups) : $attribute_groups;
+            }
+          }
           break;
 
         case 'external':
