diff --git a/advagg.drush.inc b/advagg.drush.inc
index 1bc8028..1a4dd9d 100644
--- a/advagg.drush.inc
+++ b/advagg.drush.inc
@@ -133,10 +133,10 @@ function drush_advagg_clear_all_files() {
   // Clear out the cache.
   Cache::invalidateTags(['library_info']);
   \Drupal::cache('advagg')->invalidateAll();
-  $pub = $file_system->realpath('public://');
+  $pub = $file_system->realpath('assets://');
   $css_count = count(glob($pub . '/css/optimized/*.css'));
   $js_count = count(glob($pub . '/js/optimized/*.js'));
-  foreach (['public://js/optimized', 'public://css/optimized'] as $path) {
+  foreach (['assets://js/optimized', 'assets://css/optimized'] as $path) {
     if (file_exists($path)) {
       $file_system->deleteRecursive($path);
     }
diff --git a/advagg.install b/advagg.install
index 80c6b00..b8146ce 100644
--- a/advagg.install
+++ b/advagg.install
@@ -14,17 +14,17 @@ use Drupal\advagg\Form\OperationsForm;
  */
 function advagg_install() {
   // Make sure permissions for dirs are correct. Needed if installed via drush.
-  $stat_public = stat('public://');
+  $stat_public = stat('assets://');
   // Check if this folder already exists - could be the case on re-install.
-  if (!file_exists('public://css')) {
-    \Drupal::service('file_system')->mkdir('public://css');
+  if (!file_exists('assets://css')) {
+    \Drupal::service('file_system')->mkdir('assets://css');
   }
-  $stat_css = stat('public://css');
+  $stat_css = stat('assets://css');
   // Check if this folder already exists - could be the case on re-install.
-  if (!file_exists('public://js')) {
-    \Drupal::service('file_system')->mkdir('public://js');
+  if (!file_exists('assets://js')) {
+    \Drupal::service('file_system')->mkdir('assets://js');
   }
-  $stat_js = stat('public://js');
+  $stat_js = stat('assets://js');
   if (isset($stat_public['uid'])) {
     if (isset($stat_css['uid']) && $stat_public['uid'] != $stat_css['uid']) {
       @chown($stat_css[0], $stat_public['uid']);
diff --git a/advagg.module b/advagg.module
index e8c2870..1b2062c 100644
--- a/advagg.module
+++ b/advagg.module
@@ -64,7 +64,7 @@ function advagg_cron($bypass = FALSE) {
   $stale_file_threshold = \Drupal::config('system.performance')->get('stale_file_threshold');
 
   foreach (['css', 'js'] as $type) {
-    $path = $file_system->realpath("public://{$type}/optimized");
+    $path = $file_system->realpath("assets://{$type}/optimized");
     if ($files = glob("{$path}/*.{$type}")) {
       foreach ($files as $file) {
         // Only delete files older than stale file threshold defined under Cron Options.
diff --git a/advagg_ext_minify/src/Asset/Minifier.php b/advagg_ext_minify/src/Asset/Minifier.php
index 2514467..9867e91 100644
--- a/advagg_ext_minify/src/Asset/Minifier.php
+++ b/advagg_ext_minify/src/Asset/Minifier.php
@@ -59,8 +59,8 @@ class Minifier extends SingleAssetOptimizerBase {
     parent::__construct($logger, $file_url_generator);
     $this->config = $config_factory->get('advagg_ext_minify.settings');
     $this->file = $file;
-    $this->in = $file->realpath($file->tempnam('public://js/optimized', 'advagg_in'));
-    $this->out = $file->realpath($file->tempnam('public://js/optimized', 'advagg_out'));
+    $this->in = $file->realpath($file->tempnam('assets://js/optimized', 'advagg_in'));
+    $this->out = $file->realpath($file->tempnam('assets://js/optimized', 'advagg_out'));
     $this->root = $root;
     $this->fileUrlGenerator = $file_url_generator;
   }
diff --git a/src/Asset/AssetOptimizer.php b/src/Asset/AssetOptimizer.php
index 776c0ea..d9f5936 100644
--- a/src/Asset/AssetOptimizer.php
+++ b/src/Asset/AssetOptimizer.php
@@ -281,7 +281,7 @@ abstract class AssetOptimizer {
     // Prefix filename to prevent blocking by firewalls which reject files
     // starting with "ad*".
     // Create the css/ or js/ path within the files folder.
-    $path = 'public://' . $this->extension . '/optimized';
+    $path = 'assets://' . $this->extension . '/optimized';
     $version = Crypt::hashBase64($data);
     $uri = "{$path}/{$this->extension}_{$cid}.{$version}.{$this->extension}";
 
@@ -477,7 +477,7 @@ abstract class AssetOptimizer {
    */
   public static function generateHtaccess($extension, $regenerate = FALSE) {
     $fileSystem = \Drupal::service('file_system');
-    $path = "public://{$extension}/optimized";
+    $path = "assets://{$extension}/optimized";
     $file = $path . '/.htaccess';
     if (!$regenerate && file_exists($file)) {
       return;
diff --git a/src/Commands/AdvaggCommands.php b/src/Commands/AdvaggCommands.php
index 663e287..b38328b 100644
--- a/src/Commands/AdvaggCommands.php
+++ b/src/Commands/AdvaggCommands.php
@@ -111,10 +111,10 @@ class AdvaggCommands extends DrushCommands {
     // Clear out the cache.
     Cache::invalidateTags(['library_info']);
     $this->cache->invalidateAll();
-    $pub = $this->fileSystem->realpath('public://');
+    $pub = $this->fileSystem->realpath('assets://');
     $css_count = count(glob($pub . '/css/optimized/*.css'));
     $js_count = count(glob($pub . '/js/optimized/*.js'));
-    foreach (['public://js/optimized', 'public://css/optimized'] as $path) {
+    foreach (['assets://js/optimized', 'assets://css/optimized'] as $path) {
       if (file_exists($path)) {
         $this->fileSystem->deleteRecursive($path);
       }
diff --git a/src/Form/InfoForm.php b/src/Form/InfoForm.php
index 83c736a..912dbcd 100644
--- a/src/Form/InfoForm.php
+++ b/src/Form/InfoForm.php
@@ -234,7 +234,7 @@ class InfoForm extends ConfigFormBase {
    */
   private function getRandomFile() {
     // Ensure the directory exists.
-    $dir = 'public://js/optimized/';
+    $dir = 'assets://js/optimized/';
     if (!is_dir($dir)) {
       return FALSE;
     }
diff --git a/src/Form/OperationsForm.php b/src/Form/OperationsForm.php
index d3c673a..b05605b 100644
--- a/src/Form/OperationsForm.php
+++ b/src/Form/OperationsForm.php
@@ -184,10 +184,10 @@ class OperationsForm extends ConfigFormBase {
     // Clear out the cache.
     Cache::invalidateTags(['library_info']);
     $this->cache->invalidateAll();
-    $pub = $this->fileSystem->realpath('public://');
+    $pub = $this->fileSystem->realpath('assets://');
     $css_count = count(glob($pub . '/css/optimized/*.css'));
     $js_count = count(glob($pub . '/js/optimized/*.js'));
-    foreach (['public://js/optimized', 'public://css/optimized'] as $path) {
+    foreach (['assets://js/optimized', 'assets://css/optimized'] as $path) {
       if (file_exists($path)) {
         $this->fileSystem->deleteRecursive($path);
       }
