diff --git a/permissions_by_term.module b/permissions_by_term.module
index 4b3f8d9da1d6249a32a07f885a56d3f16ae774e1..e362428c5798938598ea34e1018aa7c0f47a9107 100644
--- a/permissions_by_term.module
+++ b/permissions_by_term.module
@@ -348,9 +348,11 @@ function permissions_by_term_form_alter(&$form, FormStateInterface $formState, $
      */
     $nodeEntityBundleInfo = \Drupal::service('permissions_by_term.node_entity_bundle_info');
 
+    $nodeDetails = $nodeEntityBundleInfo->prepareNodeDetails($langcode, $nid);
     $form['permissions_by_term_info']['revision'] = array(
-      '#type' => 'item',
-      '#markup' => $nodeEntityBundleInfo->renderNodeDetails($viewFilePath, $langcode, $nid),
+      '#theme' => 'permissions_by_term_render_node_details',
+      '#roles' => $nodeDetails['roles'],
+      '#users' => $nodeDetails['users'],
     );
 
     $form['#attached']['library'][] = 'permissions_by_term/nodeForm';
@@ -563,3 +565,20 @@ function permissions_by_term_taxonomy_term_delete(EntityInterface $entity) {
   $cacheInvalidator = \Drupal::service('permissions_by_term.cache_invalidator');
   $cacheInvalidator->invalidate();
 }
+
+/**
+ * Implements hook_theme().
+ */
+function permissions_by_term_theme() {
+  return [
+    'permissions_by_term_render_node_details' => [
+      'variables' => [
+        'roles' => [],
+        'users' => [],
+      ],
+      'template' => 'node-details',
+      'path' => \Drupal::service('extension.list.module')->getPath('permissions_by_term') . '/src/View',
+    ]
+  ];
+}
+
diff --git a/src/Service/NodeEntityBundleInfo.php b/src/Service/NodeEntityBundleInfo.php
index e9b5f31515fbdb59985bcddf2f3a14c6831173f5..ae7b8ef21ca49f7ad37ed446a6e06cc6e4a3d0fb 100644
--- a/src/Service/NodeEntityBundleInfo.php
+++ b/src/Service/NodeEntityBundleInfo.php
@@ -55,16 +55,15 @@ class NodeEntityBundleInfo {
   }
 
   /**
-   * @param $viewFilePath
-   * @param null $nid
    * @param string $langcode
+   * @param null $nid
    *
-   * @return string
+   * @return array
    * @throws \Twig_Error_Loader
    * @throws \Twig_Error_Runtime
    * @throws \Twig_Error_Syntax
    */
-  public function renderNodeDetails($viewFilePath, $langcode, $nid = FALSE) {
+  public function prepareNodeDetails($langcode, $nid = FALSE) {
     $roles = null;
     $users = null;
     $rids = null;
@@ -86,13 +85,10 @@ class NodeEntityBundleInfo {
       $users = User::loadMultiple($uids);
     }
 
-    $templateClass = $this->twig->getTemplateClass($viewFilePath);
-    $template = $this->twig->loadTemplate($templateClass, $viewFilePath);
-
-    return $template->render(['roles' => $roles, 'users' => $users]);
-
+    return ['roles' => $roles, 'users' => $users];
   }
 
+
   /**
    * @return array
    */
