diff --git a/permissions_by_term.install b/permissions_by_term.install
index d26cc46..498ef02 100644
--- a/permissions_by_term.install
+++ b/permissions_by_term.install
@@ -89,15 +89,6 @@ function permissions_by_term_requirements(string $phase) {
         'value'       => t('No dynamic page cache'),
       ];
     }
-
-    if (!\Drupal::moduleHandler()->moduleExists('page_cache')) {
-      $requirements['permissions_by_term_page_cache_missing'] = [
-        'severity'    => REQUIREMENT_WARNING,
-        'title'       => t('Permissions by Term: page cache module is not installed.'),
-        'description' => t("The 'Page Cache' module is not installed. Without this module your site's performance will be slow. Ensure having the dynamic page cache module and page cache module installed."),
-        'value'       => t('No page cache'),
-      ];
-    }
   }
 
   return $requirements;
