diff --git a/src/Commands/WarmerCommands.php b/src/Commands/WarmerCommands.php
index eed20b1..79f8edb 100644
--- a/src/Commands/WarmerCommands.php
+++ b/src/Commands/WarmerCommands.php
@@ -111,7 +111,7 @@ class WarmerCommands extends DrushCommands {
       return;
     }
     $this->logger()->success(dt('Warming caches in @count batches from the "warmer" queue.', ['@count' => $batch_count]));
-    $this->queueCommands->run('warmer', ['time-limit' => static::VERY_HIGH_NUMBER]);
+    $this->queueCommands->run('warmer', ['time-limit' => static::VERY_HIGH_NUMBER, 'items-limit' => $total]);
   }
 
   /**
