diff --git a/src/Checks/FilePermissions.php b/src/Checks/FilePermissions.php
index 64dc022dce5d2e99104f0e0526a9e057a927a6b0..9c68feb6a6e2ea2837bfdb35ae81096c369e5193 100644
--- a/src/Checks/FilePermissions.php
+++ b/src/Checks/FilePermissions.php
@@ -284,6 +284,21 @@ HTML;
       $ignore[] = $private_files;
     }
 
+    // If the assets stream wrapper service exists, get the assets path.
+    //
+    // @see https://www.drupal.org/project/drupal/issues/3027639
+    //   Drupal core issue to add this. Planned to be released in Drupal core
+    //   10.1.
+    //
+    // @todo Inject this into the constructor?
+    if ($this->container->has('stream_wrapper.assets')) {
+
+      $assetsPath = $this->container->get('stream_wrapper.assets')->basePath();
+
+      $ignore[] = \realpath($assetsPath);
+
+    }
+
     $this->moduleHandler()->alter('security_review_file_ignore', $ignore);
     return $ignore;
   }
