diff --git a/src/ProcessBase.php b/src/ProcessBase.php
index 40f8cde..6d939be 100644
--- a/src/ProcessBase.php
+++ b/src/ProcessBase.php
@@ -158,8 +158,6 @@ class ProcessBase extends Process
             // Doubled double quotes were converted to \\".
             // Revert to double quote.
             $output = str_replace('\\"', '"', $output);
-            // Revert of doubled backslashes.
-            $output = preg_replace('#\\\\{2}#', '\\', $output);
         }
         $sanitizedOutput = $this->removeNonJsonJunk($output);
         $json = json_decode($sanitizedOutput, true);
